config.go

```diff
@@ -0,0 +1,213 @@
+	// The default is standard error.
+	ErrorOutputPaths []string `json:"error_output_paths",yaml:"error_output_paths"`
+	// InitialFields is a collection of fields to add to the root logger.
+	InitialFields map[string]interface{} `json:"initial_fields",yaml:"initial_fields"`
+}
+
+// NewProductionConfig is the recommended production configuration. Logging is
+// enabled at InfoLevel and above.
+//
+// It uses a JSON encoder, writes logs to stdout and internal errors to stderr,
+// and enables sampling. Stacktraces are automatically included on logs of
+// ErrorLevel and above.
+func NewProductionConfig() Config {
+	return Config{
+		Level:       DynamicLevel(),
+		Development: false,
+		Sampling: &SamplingConfig{
+			Initial:    100,
+			Thereafter: 100,
+		},
+		Encoding: "json",
+		EncoderConfig: zapcore.EncoderConfig{
+			TimeKey:        "ts",
+			LevelKey:       "level",
+			NameKey:        "logger",
+			CallerKey:      "caller",
+			MessageKey:     "msg",
+			StacktraceKey:  "stacktrace",
+			EncodeLevel:    zapcore.LowercaseLevelEncoder,
+			EncodeTime:     zapcore.EpochTimeEncoder,
+			EncodeDuration: zapcore.SecondsDurationEncoder,
+		},
+		OutputPaths:      []string{"stdout"},
+		ErrorOutputPaths: []string{"stderr"},
+	}
+}
+
+// NewDevelopmentConfig is a reasonable development configuration. Logging is
+// enabled at DebugLevel and above.
+//
+// It (obviously) enables development mode, uses a console encoder, writes logs
+// to stdout and internal errors to stderr, and disables sampling. Stacktraces
+// are automatically included on logs of WarnLevel and above.
+func NewDevelopmentConfig() Config {
+	dyn := DynamicLevel()
+	dyn.SetLevel(DebugLevel)
+
+	return Config{
+		Level:       dyn,
+		Development: true,
+		Encoding:    "console",
+		EncoderConfig: zapcore.EncoderConfig{
+			// Keys can be anything except the empty string.
+			TimeKey:        "T",
+			LevelKey:       "L",
+			NameKey:        "N",
+			CallerKey:      "C",
+			MessageKey:     "M",
+			StacktraceKey:  "S",
+			EncodeLevel:    zapcore.CapitalLevelEncoder,
+			EncodeTime:     zapcore.ISO8601TimeEncoder,
+			EncodeDuration: zapcore.StringDurationEncoder,
+		},
+		OutputPaths:      []string{"stdout"},
+		ErrorOutputPaths: []string{"stderr"},
+	}
+}
+
+// Build constructs a logger from the Config and Options.
+func (cfg Config) Build(opts ...Option) (*Logger, error) {
+	sink, errSink, err := cfg.openSinks()
+	if err != nil {
+		return nil, err
+	}
+
+	enc, err := cfg.buildEncoder()
+	if err != nil {
+		return nil, err
+	}
+
+	return New(
+		zapcore.WriterFacility(enc, sink, cfg.Level),
+		cfg.buildOptions(errSink)...,
+	).WithOptions(opts...), nil
+}
+
+func (cfg Config) buildOptions(errSink zapcore.WriteSyncer) []Option {
+	opts := []Option{ErrorOutput(errSink)}
+
+	if cfg.Development {
+		opts = append(opts, Development())
+	}
+
+	if !cfg.DisableCaller {
+		opts = append(opts, AddCaller())
+	}
+
+	stackLevel := ErrorLevel
+	if cfg.Development {
+		stackLevel = WarnLevel
```

Comment:
Since we're handling a lot of different things in here, I'd rather handle each of the config struct's fields in a separate little block.